### PR TITLE
Feature/typescript

### PR DESCRIPTION
**Pull request by Lulu-Mews**
converted main to typescript, please check if everything works correctly! thank

### CODE REVIEW

File: src/component/ProductCart/CartProduct.tsx [hunk 1]
```diff
@@ -41,13 +45,15 @@ const CartProduct = ({ product, amount }) => {
...
+          src={product.images?.[0]}
           alt=""
         />
-        <Flex width="50%">{product.description}</Flex>
+        <Flex width="50%">{product.name}</Flex>
       </Flex>
       <Flex align="end" width="20%" isVertical>
-        {priceFormatter.format(product.price * amount * (product.vat + 1))}
+        {priceFormatter.format(
+          (product.price || 0) * amount * ((product.vat || 0) + 1)
```
Comment (evavandenboogaart) [inline src/component/ProductCart/CartProduct.tsx:55]: product.price and product.vat aren't optional so there's no need to fallback to 0

**Review by evavandenboogaart — APPROVED** [review]

File: src/component/ProductCart/CartProduct.tsx (hunk 1 above)
Comment (Lulu-Mews) [reply src/component/ProductCart/CartProduct.tsx:55]: ive resolved it, i will push changes in a moment! thanks!

File: src/component/ProductOverview/ProductFilters.tsx [hunk 2]
```diff
@@ -1,28 +1,27 @@
...
 export const ProductFilters = () => {
-  const isMobile = window.visualViewport.width < 500;
+  const isMobile = (window.visualViewport?.width || 0) < 500;
   const [collapsed, setCollapsed] = useState(isMobile);
   const products = useGetProducts();
-  const categories = products.item.reduce((acc, product) => {
+  const categories = products?.reduce((acc, product) => {
     if (
       !product?.category ||
       acc.findIndex(
```
[the remark this answers is not in the document]
Comment (Lulu-Mews) [reply src/component/ProductOverview/ProductFilters.tsx:15]: fixed it, when i converted to typescript i had an error before but it doesnt show this time

File: src/component/shared/FileDropper/FileDropper.tsx [hunk 3]
```diff
@@ -1,16 +1,21 @@
-import React, { useState } from "react";
+import { DragEvent, PropsWithChildren, useState } from "react";
 import { StyledFileDropper } from "./styles";
+import { SpacingProps } from "../../../interfaces";
 
-export function FileDropper({ children, onUpload }) {
+const FileDropper: React.FC<
```
[the remark this answers is not in the document]
Comment (Lulu-Mews) [reply src/component/shared/FileDropper/FileDropper.tsx:5]: i added interfaces and moved some code over to make it more consistant!

File: src/component/shared/Search/Search.tsx [hunk 4]
```diff
@@ -9,37 +9,35 @@ import {
...
 } from "react-router-dom";
 
-function Search() {
+const Search: React.FC = () => {
   const [query] = useSearchParams();
   const [search, setSearchText] = useState(query.get("query"));
   const navigate = useNavigate();
   const clickSearch = () => {
-    const urlEncodedSearch = encodeURI(search);
+    const urlEncodedSearch = encodeURI(search || "");
```
[the remark this answers is not in the document]
Comment (Lulu-Mews) [reply src/component/shared/Search/Search.tsx:17]: i replaced it with:

```js
    if (!search) return;
    const urlEncodedSearch = encodeURI(search);
```

**Review by evavandenboogaart — APPROVED** [review]
Looks good

**Pull request merged**